streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorTopology.java

```diff
@@ -27,48 +27,109 @@
+                Collections.<String>emptySet());
+    }
+
+    public static ProcessorTopology withRepartitionTopics(final List<ProcessorNode> processorNodes,
+                                                          final Map<String, SourceNode> sourcesByTopic,
+                                                          final Set<String> repartitionTopics) {
+        return new ProcessorTopology(processorNodes,
+                sourcesByTopic,
+                Collections.<String, SinkNode>emptyMap(),
+                Collections.<StateStore>emptyList(),
+                Collections.<StateStore>emptyList(),
+                Collections.<String, String>emptyMap(),
+                repartitionTopics);
+    }
 
     public ProcessorTopology(final List<ProcessorNode> processorNodes,
-                             final Map<String, SourceNode> sourceByTopics,
-                             final Map<String, SinkNode> sinkByTopics,
+                             final Map<String, SourceNode> sourcesByTopic,
+                             final Map<String, SinkNode> sinksByTopic,
                              final List<StateStore> stateStores,
-                             final Map<String, String> storeToChangelogTopic,
-                             final List<StateStore> globalStateStores) {
+                             final List<StateStore> globalStateStores,
+                             final Map<String, String> stateStoreToChangelogTopic,
+                             final Set<String> repartitionTopics) {
         this.processorNodes = Collections.unmodifiableList(processorNodes);
-        this.sourceByTopics = Collections.unmodifiableMap(sourceByTopics);
-        this.sinkByTopics   = Collections.unmodifiableMap(sinkByTopics);
-        this.stateStores    = Collections.unmodifiableList(stateStores);
-        this.storeToChangelogTopic = Collections.unmodifiableMap(storeToChangelogTopic);
+        this.sourcesByTopic = Collections.unmodifiableMap(sourcesByTopic);
+        this.sinksByTopic = Collections.unmodifiableMap(sinksByTopic);
+        this.stateStores = Collections.unmodifiableList(stateStores);
         this.globalStateStores = Collections.unmodifiableList(globalStateStores);
+        this.storeToChangelogTopic = Collections.unmodifiableMap(stateStoreToChangelogTopic);
+        this.repartitionTopics = Collections.unmodifiableSet(repartitionTopics);
     }
 
     public Set<String> sourceTopics() {
-        return sourceByTopics.keySet();
+        return sourcesByTopic.keySet();
     }
 
     public SourceNode source(String topic) {
-        return sourceByTopics.get(topic);
+        return sourcesByTopic.get(topic);
     }
 
     public Set<SourceNode> sources() {
-        return new HashSet<>(sourceByTopics.values());
+        return new HashSet<>(sourcesByTopic.values());
     }
 
     public Set<String> sinkTopics() {
-        return sinkByTopics.keySet();
+        return sinksByTopic.keySet();
```

Comment:
not part of this PR but this method doesn't appear to be used

Reply:
Yup I realized that as well, left it as is intentionally for future uses